python/pyarrow/types.pxi

```diff
@@ -1615,6 +1615,44 @@ cdef class Schema:
         return self.__str__()
 
 
+def unify_schemas(list schemas):
+    """
+    Unifies schemas by merging fields by name.
+
+    The resulting schema will contain the union of fields from all schemas.
+    Fields with the same name will be merged.
+
+    - The unified field will inherit the metadata from the schema where
+        that field is first defined.
```

Comment:
I suppose it would be better if metadata was merged too, but that's probably on the C++ side....